wallet/src/account/mod.rs

```diff
@@ -981,14 +983,35 @@ impl<K: AccountKeyChains> Account<K> {
             outputs.push(TxOutput::Transfer(output_value, output_destination));
         }
 
-        let nonce = order_info
-            .nonce
-            .map_or(Some(AccountNonce::new(0)), |n| n.increment())
-            .ok_or(WalletError::OrderNonceOverflow(order_id))?;
-        let request = SendRequest::new().with_outputs(outputs).with_inputs_and_destinations([(
-            TxInput::AccountCommand(nonce, AccountCommand::ConcludeOrder(order_id)),
-            order_info.conclude_key.clone(),
-        )]);
+        let version = self
+            .chain_config
+            .chainstate_upgrades()
+            .version_at_height(self.account_info.best_block_height())
+            .1
+            .orders_version();
+
```

Comment:
Tecnically, this can be "best_block_height + 1", because when the tx is mined, the upgrade height will be reached. (Unless a reorg occurs and the tx gets mined at a lower height, but I guess this is always possible).